pipeline/kafka-connector/src/main/scala/org/sunbird/obsrv/kafkaconnector/task/KafkaConnectorStreamTask.scala

```diff
@@ -0,0 +1,71 @@
+package org.sunbird.obsrv.kafkaconnector.task
+
+import com.typesafe.config.ConfigFactory
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.api.java.typeutils.TypeExtractor
+import org.apache.flink.api.java.utils.ParameterTool
+import org.apache.flink.streaming.api.datastream.{DataStream, SingleOutputStreamOperator}
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment
+import org.sunbird.obsrv.core.streaming.{BaseStreamTask, FlinkKafkaConnector}
+import org.sunbird.obsrv.core.util.FlinkUtil
+import org.sunbird.obsrv.registry.DatasetRegistry
+
+import java.io.File
+import scala.collection.mutable
+
+class KafkaConnectorStreamTask(config: KafkaConnectorConfig, kafkaConnector: FlinkKafkaConnector) extends BaseStreamTask[mutable.Map[String, AnyRef]] {
+
+  private val serialVersionUID = -7729362727131516112L
+
+  // $COVERAGE-OFF$ Disabling scoverage as the below code can only be invoked within flink cluster
+  def process(): Unit = {
+    implicit val env: StreamExecutionEnvironment = FlinkUtil.getExecutionContext(config)
+    implicit val mapTypeInfo: TypeInformation[mutable.Map[String, AnyRef]] = TypeExtractor.getForClass(classOf[mutable.Map[String, AnyRef]])
+    implicit val stringTypeInfo: TypeInformation[String] = TypeExtractor.getForClass(classOf[String])
+
+    val datasetSourceConfig = DatasetRegistry.getDatasetSourceConfig()
+    datasetSourceConfig.map { configList =>
+      val dataStreamList = configList.filter(_.connectorType.equalsIgnoreCase("kafka")).map {
+        dataSourceConfig =>
+          val dataStream: DataStream[mutable.Map[String, AnyRef]] =
+            getMapDataStream(env, config, List(dataSourceConfig.connectorConfig.topic),
+            config.kafkaConsumerProperties(kafkaBrokerServers = Some(dataSourceConfig.connectorConfig.kafkaBrokers),
+              kafkaConsumerGroup = Some(s"kafka-${dataSourceConfig.connectorConfig.topic}")),
+              s"kafka-${dataSourceConfig.connectorConfig.topic}", kafkaConnector)
+          val datasetId = dataSourceConfig.datasetId
+          val resultMapStream: DataStream[mutable.Map[String, AnyRef]] = dataStream.map {
+            streamMap: mutable.Map[String, AnyRef] => {
+              val result: mutable.Map[String, AnyRef] = streamMap + ("datasetId" -> datasetId)
+              result
+            }
+          }.returns(classOf[mutable.Map[String, AnyRef]])
+          print(resultMapStream)
+          resultMapStream.sinkTo(kafkaConnector.kafkaMapSink(config.kafkaOutputTopic))
```

Comment:
Master dataset should sink to different topic. The inputTopic can be fetched from datasetConfig

Reply:
The output topic is now derived from the dataset_config for a dataset.